stdlib/map.gr

```diff
@@ -8,115 +8,111 @@ import WasmI32 from "runtime/unsafe/wasmi32"
 import { allocateArray } from "runtime/dataStructures"
 
 // TODO: Consider implementing this as List<(Box<k>, Box<v>)>
-record Bucket<k, v> {
-  mut key: k,
-  mut value: v,
-  mut next: Option<Bucket<k, v>>
-}
+record Bucket<k, v> { mut key: k, mut value: v, mut next: Option<Bucket<k, v>> }
 
-record Map<k, v> {
-  mut size: Number,
-  mut buckets: Array<Option<Bucket<k, v>>>
-}
+record Map<k, v> { mut size: Number, mut buckets: Array<Option<Bucket<k, v>>> }
 
 // TODO: This could take an `eq` function to custom comparisons
-export let makeSized = (size) => {
-  let buckets = Array.make(size, None);
-  {
-    size: 0,
-    buckets
-  }
+export let makeSized = size => {
+  let buckets = Array.make(size, None)
+  { size: 0, buckets }
 }
 
 export let make = () => {
   makeSized(16)
 }
 
 let getBucketIndex = (key, buckets) => {
-  let bucketsLength = Array.length(buckets);
-  let hashedKey = hash(key);
+  let bucketsLength = Array.length(buckets)
+  let hashedKey = hash(key)
   hashedKey % bucketsLength
 }
 
 let rec copyNodeWithNewHash = (oldNode, next, tail) => {
   match (oldNode) {
     None => void,
     Some(node) => {
-      let idx = getBucketIndex(node.key, next);
-      let newNode = Some(node);
+      let idx = getBucketIndex(node.key, next)
+      let newNode = Some(node)
       match (tail[idx]) {
         None => {
-          next[idx] = newNode;
+          next[idx] = newNode
         },
         Some(tailNode) => {
           // If there's already a tail node, we add this to the end
-          tailNode.next = newNode;
-        }
+          tailNode.next = newNode
+        },
       }
       // Always place this node as the new tail
-      tail[idx] = newNode;
+      tail[idx] = newNode
       // Recurse with the next node
-      copyNodeWithNewHash(node.next, next, tail);
-    }
+      copyNodeWithNewHash(node.next, next, tail)
+    },
   }
 }
 
-let resize = (map) => {
-  let currentBuckets = map.buckets;
-  let currentSize = Array.length(currentBuckets);
-  let nextSize = currentSize * 2;
+let resize = map => {
+  let currentBuckets = map.buckets
+  let currentSize = Array.length(currentBuckets)
+  let nextSize = currentSize * 2
   if (nextSize >= currentSize) {
-    let nextBuckets = Array.make(nextSize, None);
+    let nextBuckets = Array.make(nextSize, None)
     // This tracks the tail nodes so we can set their `next` to None
-    let tailNodes = Array.make(nextSize, None);
-    map.buckets = nextBuckets;
-    Array.forEach((old) => {
-      copyNodeWithNewHash(old, nextBuckets, tailNodes);
-    }, currentBuckets);
-    Array.forEach((tail) => {
-      match (tail) {
-        None => void,
-        Some(node) => {
-          node.next = None;
+    let tailNodes = Array.make(nextSize, None)
+    map.buckets = nextBuckets
+    Array.forEach(
+      old => {
+        copyNodeWithNewHash(old, nextBuckets, tailNodes)
+      },
+      currentBuckets,
+    )
```

Comment:
Yeah it seems like arguments that are functions always get pushed. This is fairly common, so do you think there's anything we can do about it?

Reply:
The rules of the formatter engine we are using is that once one thing breaks, everything should break.  And the pexpblock forces a break for the new lines inside the braces block.    The reason reformatter seems to do the same.